src/internal/m365/collection/exchange/collection.go

```diff
@@ -174,18 +176,46 @@ func NewCollection(
 	bc baseCollection,
 	user string,
 	items itemGetterSerializer,
+	added map[string]time.Time,
+	removed []string,
+	validModTimes bool,
 	statusUpdater support.StatusUpdater,
-) prefetchCollection {
-	collection := prefetchCollection{
+) data.BackupCollection {
+	add := maps.Clone(added)
+	remove := make(map[string]struct{}, len(removed))
```

Comment:
nit: Since they are only used once, why not rename them to a bigger ones and use normal names for the vars used within the func.

```suggestion
	added := maps.Clone(origAdded)
	removed := make(map[string]struct{}, len(origRemoved))
```